cpp/benchmarks/groupby/group_nunique.cpp

```diff
@@ -39,17 +39,23 @@ auto make_aggregation_request_vector(cudf::column_view const& values, Args&&...
 template <typename Type>
 void bench_groupby_nunique(nvbench::state& state, nvbench::type_list<Type>)
 {
-  auto const size = static_cast<cudf::size_type>(state.get_int64("num_rows"));
+  auto const size        = static_cast<cudf::size_type>(state.get_int64("num_rows"));
+  auto const cardinality = static_cast<cudf::size_type>(state.get_int64("cardinality"));
 
   auto const keys = [&] {
-    data_profile profile = data_profile_builder().cardinality(0).no_validity().distribution(
-      cudf::type_to_id<int32_t>(), distribution_id::UNIFORM, 0, 100);
+    data_profile profile =
+      data_profile_builder()
+        .cardinality(cardinality)
+        .no_validity()
+        .distribution(cudf::type_to_id<int32_t>(), distribution_id::UNIFORM, 0, size);
```

Comment:
Code to create `keys` and `values` here seem to be very similar (the same) as for creating those of groupby max. Can we further extract them into a common function? Like `create_keys_values()`?

Reply:
The duplicate part is just data profile construction. Though we repeat those 5 lines of code each time, it's just one API call. It's not worth creating a helper function wrapping one single call IMO.